internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -439,13 +440,29 @@ func (m DeploymentResource) Create() sdk.ResourceFunc {
 			}
 
 			isBasicSKU := strings.HasPrefix(model.Sku, "basic")
-			if isBasicSKU && (model.Capacity != defaultCapacity || len(model.AutoScaleProfile) > 0) {
-				return fmt.Errorf("basic SKUs are incompatible with `capacity` or `auto_scale_profiles`")
-			}
+			if !features.FourPointOhBeta() {
+				if isBasicSKU && (model.Capacity != defaultCapacity || len(model.AutoScaleProfile) > 0) {
+					return fmt.Errorf("basic SKUs are incompatible with `capacity` or `auto_scale_profiles`")
+				}
 
-			if model.Capacity > 0 && !isBasicSKU {
-				prop.ScalingProperties = &nginxdeployment.NginxDeploymentScalingProperties{
-					Capacity: pointer.FromInt64(model.Capacity),
+				if model.Capacity > 0 && !isBasicSKU {
+					prop.ScalingProperties = &nginxdeployment.NginxDeploymentScalingProperties{
+						Capacity: pointer.FromInt64(model.Capacity),
+					}
+				}
+			} else {
+				hasScaling := (model.Capacity > 0 || len(model.AutoScaleProfile) > 0)
+				if isBasicSKU && hasScaling {
+					return fmt.Errorf("basic SKUs are incompatible with `capacity` or `auto_scale_profiles`")
```

Comment:
Do we need to error here or do we want our API/future dry-run to handle this validation?

Reply:
I think we want the error here for now, and I'm gonna add it to the update function as well. You're right that the API handles this, but until that "dry-run" feature gets implemented server side we can end up with some bad data. I figured this validation is unlikely to change, so OK to duplicate between client and server.